### ISSUE TITLE
Build ERROR after fixing maybe-uninitialized warning

### ISSUE THREAD

**Issue by pmajchrzak** [PaddlePaddle/Paddle#43156]
### 问题描述 Issue Description

Commit f1f79b0d9d18cebcf8b89775d2b066d6fdd04199 causes build error, reproduced on several machines (SKX, CLX, CPX)

![image](https://user-images.githubusercontent.com/10847947/171390002-227e30d5-8348-488f-927d-0255561a901a.png)
↑↑↑
These errors show up in many places

### PaddlePaddle branch: 
develop

### CMAKE Flags:
``
-DCMAKE_BUILD_TYPE=Release -DWITH_GPU=OFF -DWITH_MKLDNN=ON -DWITH_TESTING=ON -DWITH_PROFILER=OFF -DWITH_STYLE_CHECK=OFF -DON_INFER=ON -DWITH_INFERENCE_API_TEST=ON -DINFERENCE_DEMO_INSTALL_DIR=/data/PaddlePaddle/pp_models -DPY_VERSION=3.6
``
### Environment
```
            OS: Ubuntu18.04 with mkldnn 
 cmake version: 3.10.2
      Compiler: GCC 8
```

**Comment by zhiqiu** [comment]
Hi, @pmajchrzak 
For this case, change `auto in_dims = x.dims()` to `const auto & in_dims = x.dims()` can solve the problem. But it is strange that I can build these file on my machine and on CI machine.

**Comment by pmajchrzak** [comment]
Thanks :)
I am only a modest validation soldier, so I am rather not the addressee of this solution.
Someone from the development team should probably fix this error.

**Comment by pmajchrzak** [comment]
The compiler version has been updated to gcc 9.4.0 and unfortunately, the bug still exists. 
![image](https://user-images.githubusercontent.com/10847947/173043667-c40f2000-c6c9-41f0-bf75-034406315478.png)


BTW CMake communicating that version 9.4 is too high in turn.
![image](https://user-images.githubusercontent.com/10847947/173043553-a9d4fd16-5b60-4d36-b79d-dfe2808ea22a.png)

**Comment by zhiqiu** [comment]
Hi, could you please give us an enviroment, like docker image, to reproduce and fix the problem?

update, the problem is fixed in #43511
